lib/logstash/inputs/snmptrap.rb

```diff
@@ -47,26 +47,44 @@ def register
     end
   end # def register
 
-  public
   def run(output_queue)
     begin
       # snmp trap server
       snmptrap_listener(output_queue)
     rescue => e
       @logger.warn("SNMP Trap listener died", :exception => e, :backtrace => e.backtrace)
-      sleep(5)
-      retry
+      stoppable_call(20) { sleep(0.25) }
```

Comment:
the `stop` method is called from the pipeline thread and the plugin `run` method is running in the plugin worker thread and we don't have a consistent way of retrieving the plugin worker thread. we have 2 options: 
1- use case-by-case solutions doing something like `@plugin_thread = Thread.current` when entering the `run` method. 
2- introduce a refactor to "correctly" expose the plugin thread in a consistent way across plugins.